Source/SwiftLintFramework/Rules/Style/UnneededParenthesesInClosureArgumentRule.swift

```diff
@@ -60,75 +75,74 @@ public struct UnneededParenthesesInClosureArgumentRule: ConfigurationProviderRul
     }
 
-    public func correct(file: SwiftLintFile) -> [Correction] {
-        let violatingRanges = file.ruleEnabled(violatingRanges: violationRanges(file: file), for: self)
-        var correctedContents = file.contents
-        var adjustedLocations = [Int]()
-
-        for violatingRange in violatingRanges.reversed() {
-            let correctingRange = NSRange(location: violatingRange.location + 1,
-                                          length: violatingRange.length - 2)
-            if let indexRange = correctedContents.nsrangeToIndexRange(violatingRange),
-                let updatedRange = correctedContents.nsrangeToIndexRange(correctingRange) {
-                var updatedArguments = correctedContents[updatedRange]
-                if let whiteSpaceIndex = correctedContents.index(indexRange.upperBound,
-                                                                 offsetBy: 0,
-                                                                 limitedBy: correctedContents.endIndex),
-                   !String(correctedContents[whiteSpaceIndex]).hasTrailingWhitespace() {
-                    updatedArguments += " "
-                }
-                correctedContents = correctedContents.replacingCharacters(in: indexRange,
-                                                                          with: String(updatedArguments))
-                adjustedLocations.insert(violatingRange.location, at: 0)
-            }
+    override func visit(_ node: ClosureSignatureSyntax) -> Syntax {
+        guard let clause = node.input?.as(ParameterClauseSyntax.self),
+              !clause.parameterList.contains(where: { $0.type != nil }),
+              clause.parameterList.isNotEmpty else {
+            return super.visit(node)
         }
 
-        file.write(correctedContents)
+        let isInDisabledRegion = disabledRegions.contains { region in
+            region.contains(node.positionAfterSkippingLeadingTrivia, locationConverter: locationConverter)
+        }
 
-        return adjustedLocations.map {
-            Correction(ruleDescription: Self.description,
-                       location: Location(file: file, characterOffset: $0))
+        guard !isInDisabledRegion else {
+            return super.visit(node)
         }
+
+        let items = clause.parameterList.enumerated().compactMap { idx, param -> ClosureParamSyntax? in
+            guard let name = param.firstName else {
+                return nil
+            }
+
+            let isLast = idx == clause.parameterList.count - 1
+            return SyntaxFactory.makeClosureParam(
+                name: name,
+                trailingComma: isLast ? nil : SyntaxFactory.makeCommaToken(trailingTrivia: .spaces(1))
+            )
+        }
+
+        correctionPositions.append(clause.positionAfterSkippingLeadingTrivia)
+
+        let paramList = SyntaxFactory.makeClosureParamList(items).withTrailingTrivia(.spaces(1))
+        return Syntax(node.withInput(Syntax(paramList)))
```

Comment:
Should this be `return super.visit(...)`? I don't think the superclasses ever do anything here, so I think we can just always return in this func?

Reply:
Calling the superclass is important so it does visit the node's children. In this case I don't think it matters, but decided to call `super.visit` for consistency

Reply:
Oh I think I might need to fix some rules I’ve written based on this comment!